### PR TITLE
Add public channel endpoints and tests

### PR DESCRIPTION
**Pull request by ralphiee22**
## Summary

Adds public endpoints which are identical to the ones on studio. With these endpoints another device can query the channels that are on this kolibri device.

## TODO

- [x] Have tests been written for the new code?
- [ ] Has documentation been written/updated?
- [ ] New dependencies (if any) added to requirements file
- [ ] Add an entry to CHANGELOG.rst
- [ ] Add yourself it AUTHORS.rst if you don't appear there

## Reviewer guidance

*If you PR has a significant size, give the reviewer some helpful remarks*

## Documentation

If the PR has documentation, link the file here (either .rst in your repo or if built on Read The Docs)

### CODE REVIEW

File: kolibri/core/content/serializers.py
Comment (ralphiee22) [inline kolibri/core/content/serializers.py:164]: need to add the token fields to kolibri models.

File: kolibri/core/content/serializers.py
Comment (ralphiee22) [inline kolibri/core/content/serializers.py:137]: need to add `public` field to `ChannelMetadata` models

File: kolibri/core/content/serializers.py
Comment (ralphiee22) [inline kolibri/core/content/serializers.py:127]: these serializer fields are mapped from the conversion response
https://github.com/learningequality/kolibri/blob/develop/kolibri/core/content/api.py#L655

File: kolibri/core/content/serializers.py
Comment (jamalex) [inline kolibri/core/content/serializers.py:162]: a bit confusing -- perhaps `return None if not ...`?

Comment (ralphiee22) [inline kolibri/core/content/serializers.py:162]: not sure if this is the best way to convert the time to a string
cc: @rtibbles

File: kolibri/core/content/serializers.py
Comment (jonboiser) [reply kolibri/core/content/serializers.py:164]: Might not be necessary for peer-to-peer. Will every channel on a peer device have `public = True` by default? In that case, the channel will appear in the channel list, and the user will probably not need to enter a token.

Comment (ralphiee22) [reply kolibri/core/content/serializers.py:164]: yeah so every channel will not have public set to `True`. Some content creators would like to lock down their content, hence the need for these tokens.

File: kolibri/core/content/serializers.py
Comment (rtibbles) [reply kolibri/core/content/serializers.py:162]: If the field is nullable it should convert to python from the dB properly and just be directly accessible and serializable.

Comment (ralphiee22) [reply kolibri/core/content/serializers.py:162]: I believe the problem is that it doesn't know how to serialize `last_published` because its a serializer method and it doesn't correspond to one of the model fields. It's able to serialize if its just `last_updated`

File: kolibri/core/content/utils/paths.py
Comment (jamalex) [inline kolibri/core/content/utils/paths.py:134]: Would be much cleaner to use `urlencode` here. Note that to avoid special Py2/3 logic, you can just do:
`from django.utils.http import urlencode`

```
>>> urlencode({"qqq": 7, "qaaz": "hi"})
'qqq=7&qaaz=hi'
```

File: kolibri/core/public/api.py
Comment (jamalex) [inline kolibri/core/public/api.py:44]: Don't know that we need to add the complexity of allowing for specifying other versions right now, since we don't know if we'll hit the point of needing to update that (and can add the logic in at that point). But if it's easier to leave, that's cool too.

File: kolibri/core/content/serializers.py
Comment (jamalex) [inline kolibri/core/content/serializers.py:159]: The calculated fields above all look pretty expensive, which is worrying -- could we instead precalculate some of this stuff during the metadata import process?

(can discuss -- perhaps looping @rtibbles in also)

**Review by jamalex — CHANGES_REQUESTED** [review]
Great work!

Mostly small suggestions, but main thing to discuss is efficiency of computed fields on public channel list endpoint.

File: kolibri/core/content/utils/paths.py
Comment (ralphiee22) [reply kolibri/core/content/utils/paths.py:134]: cool!

File: kolibri/core/content/serializers.py
Comment (rtibbles) [reply kolibri/core/content/serializers.py:159]: If were are doing this only over available files, would have to be during annotation, I think?

Comment (ralphiee22) [reply kolibri/core/content/serializers.py:159]: so what should be proposed solution?
Add fields to ChannelMetadata which will hold this data and probably add another method during annotation to fill in those fields, such as `calculate_fields`?

Comment (jamalex) [reply kolibri/core/content/serializers.py:159]: > Add fields to ChannelMetadata which will hold this data and probably add another method during annotation to fill in those fields, such as calculate_fields?

Sounds right to me. Let's get the merged in so we're not blocking, but seems important to then do the optimizations before 0.11 ships.

**Review by jamalex — APPROVED** [review]
Approving so we can move ahead, then @ralphiee22 will follow up on the remaining pieces after.

**Pull request merged**
